src/backend/OpenCL/CHIPBackendOpenCL.cc

```diff
@@ -908,6 +909,7 @@ void CL_CALLBACK pfn_notify(cl_event Event, cl_int CommandExecStatus,
 
 void CHIPQueueOpenCL::MemMap(const chipstar::AllocationInfo *AllocInfo,
                              chipstar::Queue::MEM_MAP_TYPE Type) {
+  auto MemMapEvent = static_cast<CHIPBackendOpenCL *>(Backend)->createCHIPEvent(ChipContext_);
```

Comment:
Perhaps assign the getNativePtr() results to the local variable so you don't have to have the cast and method call in all of those uses.